### PR TITLE
added trade-lifecycle

### PR DESCRIPTION
**Pull request by thehashmap**

### CODE REVIEW

File: src/serum-trade-lifecycle.yaml [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+      The Serum DEX is a program on the Solana blockchain. Solana has block times of 400ms, and can currently process roughly 50,000 transactions per second.
+      This allows Serum to handle hundreds of orders per second per market.
+      Tokens on Solana are instrumented as instances of the SPL Token Program. An Solana Program Library (SPL) token on Solana is analogous to an ERC20 token on Ethereum.
+      Now, before going to Trade Lifecycle, let's first understand some terminologies related to accounts specifically global accounts:-
+        * **Request Queue**: This account maintains all submitted but unprocessed order placement and cancellation requests.
+        * **Orderbook**: Speaks for itself. There's one account for bids and another for asks, but for simplicity,
+        we'll refer to both of them collectively as Orderbook going forward.
+        * **Event Queue**: Reports the list of outputs from order matching: trades, for instance
+        * **Market**: Holds metadata about the market (e.g. important constants like tick size and references to the other accounts below)
+        * **Base Currency Vault**: An SPL token account that holds base currency balances
```
Comment (RobinNagpal) [inline src/serum-trade-lifecycle.yaml]: What is a base currency? and what is a quote currency? May be we should define those also. 

Does a user have base currency account for each of the different type of tokens?

File: src/serum-trade-lifecycle.yaml [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+        * **Quote Currency Vault**: An SPL token account that holds quote currency balances
+
+      And the user-specific accounts: to interact with the DEX, a given user must create an **OpenOrders account**. This account stores the following:
```
Comment (RobinNagpal) [inline src/serum-trade-lifecycle.yaml]: Is "Open Orders Account" created and maintainer by the user or by the protocol or DEX?

File: src/serum-trade-lifecycle.yaml (hunk 1 above)
Comment (RobinNagpal) [reply src/serum-trade-lifecycle.yaml]: Also what is the difference in `Base Currency Vault` and `Open Order Account`?

File: src/serum-trade-lifecycle.yaml [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+          * Their SPL token account for the market's base currency (if selling) or quote currency (if buying)
+        * 2. The **DEX program** then:
+          -Transfers funds:
+            * Determines the max funds required for this order (the size if selling, or size * price if buying)
+            * Notes the free balance for the corresponding currency indicated in the OpenOrders account
+            * Transfers the difference between the required amount and free OpenOrders balance from the SPL token account to the Base Currency Vault or Quote Currency Vault.
+            * Increments the OpenOrders account's total balance for the corresponding currency by the amount transferred
+          - Increments and retrieves a sequence number from the Request Queue: this, along with the order's price, determines the new order's ID.
+          - Adds an item to the Request Queue specifying the order details (size, price, order type, side)
+          - Adds an item representing the new order to an array in the OpenOrders account in an available slot
```
Comment (RobinNagpal) [inline src/serum-trade-lifecycle.yaml]: `Adds an item representing the new order` what is the format of it?

File: src/serum-trade-lifecycle.yaml [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+        * 1. A user submits a **Place Order** instruction to the DEX program, passing:
+          * Order details: the market size, price, order type, side
+          * Their OpenOrders account on that market
+          * Their SPL token account for the market's base currency (if selling) or quote currency (if buying)
+        * 2. The **DEX program** then:
+          -Transfers funds:
+            * Determines the max funds required for this order (the size if selling, or size * price if buying)
+            * Notes the free balance for the corresponding currency indicated in the OpenOrders account
+            * Transfers the difference between the required amount and free OpenOrders balance from the SPL token account to the Base Currency Vault or Quote Currency Vault.
+            * Increments the OpenOrders account's total balance for the corresponding currency by the amount transferred
```
Comment (RobinNagpal) [inline src/serum-trade-lifecycle.yaml]: `Increments the OpenOrders account's total balance` how exactly?

File: src/serum-trade-lifecycle.yaml [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+            key: A
+          - content: User's bank account for the market
+            key: B
+          - content: User's SPL token account for the market
+            key: C
+          - content: The market size, price, order type
+            key: D
+
+      - uuid: 5ef22568-2fc3-450d-94e2-2fad26cada2e
+        content: Where does an order placement request gets added?
```
Comment (RobinNagpal) [inline src/serum-trade-lifecycle.yaml]: The same question was asked in the previous evaluation step also. We should add a different one which relates to the different account types

File: src/serum-trade-lifecycle.yaml [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+        * The request is popped off the Request Queue and processed, and then placed on the Orderbook. Any resulting trades get reported in the Event Queue.
+        * In a Match Orders transaction, clients (or crank turners: who perform the order matching process) provide a limit parameter,
+        which specifies the number of requests to process from the Request Queue.
+        -Each request is processed as follows:
+          * It is popped off of the RequestQueue account
+          * The corresponding cancellation or order placement is made on the Orderbook (the order book consists of one tree per side, and 
+          so finding the relevant part of the appropriate tree is made easier by the comparison property of Order IDs mentioned earlier)
+          * In the case of a new order that stays on the book, the Orderbook stores details about it, including the obvious specs (side, price, remaining size),
+          the public key of the order placer's OpenOrders account, and the index of this order in that OpenOrders account's order array (the slot number).
+          * Order types like IOC and post-only are enforced
```
Comment (RobinNagpal) [inline src/serum-trade-lifecycle.yaml]: We should define these `Order types like IOC and post-only are enforced`

File: src/serum-trade-lifecycle.yaml (hunk 5 above)
Comment (thehashmap) [reply src/serum-trade-lifecycle.yaml]: I have updated previous step question

**Pull request merged**